packages/core-bridge/src/lib.rs

```diff
@@ -900,8 +903,31 @@ fn worker_shutdown(mut cx: FunctionContext) -> JsResult<JsUndefined> {
     }
 }
 
+/// Convert Rust SystemTime into a JS array with 2 numbers (seconds, nanos)
+fn system_time_to_js<'a, C>(cx: &mut C, time: SystemTime) -> NeonResult<Handle<'a, JsArray>>
+where
+    C: Context<'a>,
+{
+    let nanos = time
+        .duration_since(UNIX_EPOCH)
+        .unwrap_or(Duration::ZERO)
+        .as_nanos();
+    let only_nanos = cx.number((nanos % 1_000_000_000) as f64);
+    let ts_seconds = cx.number((nanos / 1_000_000_000) as f64);
+    let ts = cx.empty_array();
+    ts.set(cx, 0, ts_seconds).unwrap();
+    ts.set(cx, 1, only_nanos).unwrap();
+    return Ok(ts);
+}
+
+/// Helper to get the current time in nanosecond resolution.
+fn get_time_of_day(mut cx: FunctionContext) -> JsResult<JsArray> {
+    system_time_to_js(&mut cx, SystemTime::now())
+}
```

Comment:
Do we want to call across FFI this often for this? Seems like it could be a bit of unneeded overhead but maybe I'm overestimating the expense

Reply:
I don't think this is much more expensive that calling `process.hrtime` or `Date.now()`, those are native functions exposed to JS as well.